star-sys/STAR/source/MmapArray.cpp

```diff
@@ -64,4 +84,13 @@ MmapArray::~MmapArray() {
         munmap(prefix_mmap, PAGE);
         prefix_mmap=NULL;
     }
+
+    if (last_page_mmap) {
+        munmap(last_page_mmap, PAGE);
+        last_page_mmap=NULL;
+    }
+
+    if (fd) {
+        close(fd);
```

Comment:
I think this should actually happen in `makeMmap`.  There's no reason to hold the fd open for the lifetime of the mmap - the man page for `mmap` clearly states
> closing the file descriptor does not unmap the region.